src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    instanceservicesessionargs *sessionargs, std::string graphID,
+    std::map<std::string, JasmineGraphInstanceService::workerPartitions> graphPartitionedHosts, int totalPartitions) {
+    int total_threads = totalPartitions;
+    std::thread *workerThreads = new std::thread[total_threads];
+    int count = 0;
+    std::map<std::string, JasmineGraphInstanceService::workerPartitions>::iterator mapIterator;
+    for (mapIterator = graphPartitionedHosts.begin(); mapIterator != graphPartitionedHosts.end(); mapIterator++) {
+        string hostName = mapIterator->first;
+        JasmineGraphInstanceService::workerPartitions workerPartitions = mapIterator->second;
+        std::vector<std::string>::iterator it;
+        for (it = workerPartitions.partitionID.begin(); it != workerPartitions.partitionID.end(); it++) {
+            workerThreads[count] =
+                std::thread(&JasmineGraphInstanceService::collectTrainedModelThreadFunction, sessionargs, hostName,
+                            workerPartitions.port, workerPartitions.dataPort, graphID, *it);
+            count++;
+        }
+    }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphVertexCount = (data);
-            graphVertexCount = Utils::trim_copy(graphVertexCount, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID:" + graphID + " Vertex Count: " + graphVertexCount, "info");
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        workerThreads[threadCount].join();
+        std::cout << "Thread [C]: " << threadCount << " joined" << std::endl;
+    }
+}
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+int JasmineGraphInstanceService::collectTrainedModelThreadFunction(instanceservicesessionargs *sessionargs,
+                                                                   std::string host, int port, int dataPort,
+                                                                   std::string graphID, std::string partition) {
+    bool result = true;
+    std::cout << pthread_self() << " host : " << host << " port : " << port << " DPort : " << dataPort << std::endl;
+    int sockfd;
+    char data[INSTANCE_DATA_LENGTH + 1];
+    bool loop = false;
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string alphaValue = (data);
-            alphaValue = Utils::trim_copy(alphaValue, " \f\n\r\t\v");
-            instance_logger.log("Received alpha: " + alphaValue, "info");
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
 
-            double alpha = std::stod(alphaValue);
+    if (sockfd < 0) {
+        std::cerr << "Cannot create socket" << std::endl;
```

Comment:
Shall we move this to a log rather than having it in `std::cerr`?

Reply:
Replaced with `instance_logger.error("Cannot create socket");` in next PR

Reply:
Ok